src/task_list_types.F

```diff
@@ -98,4 +114,35 @@ SUBROUTINE deallocate_task_list(task_list)
 
       DEALLOCATE (task_list)
    END SUBROUTINE deallocate_task_list
+
+! **************************************************************************************************
+!> \brief Grow an array of tasks while preserving the existing entries.
+!> \param tasks ...
+!> \param new_size ...
+!> \author Ole Schuett
+! **************************************************************************************************
+   SUBROUTINE reallocate_tasks(tasks, new_size)
+      TYPE(task_type), DIMENSION(:), POINTER             :: tasks
+      INTEGER, INTENT(in)                                :: new_size
+
+      INTEGER                                            :: old_size
+      TYPE(task_type), DIMENSION(:), POINTER             :: work
+
+      IF (.NOT. ASSOCIATED(tasks)) THEN
+         ALLOCATE (tasks(new_size))
+         RETURN
+      ENDIF
+
+      old_size = SIZE(tasks)
+      IF (new_size <= old_size) RETURN
+
+      ALLOCATE (work(old_size))
+      work(:) = tasks(:)
+      DEALLOCATE (tasks)
+
+      ALLOCATE (tasks(new_size))
+      tasks(1:old_size) = work(:)
+      DEALLOCATE (work)
```

Comment:
since you have pointers couldn't you allocate the new size, copy over, then re-associate the pointer? (one allocation less)